### PR TITLE
[GLUTEN-158] Support read dwrf file from hdfs

### PR DESCRIPTION
**Pull request by KevinyhZou**
## What changes were proposed in this pull request?
support to read dwrf file from hdfs as described in #388 

## How was this patch tested?
manual tests

1. read the source file from hdfs ,like
```
val lineitem = spark.read.format("dwrf").load("hdfs://testcluster/user/hive/warehouse/lineitem_dw")
```

2 .and then run a tpch scripts 
```
select
        l_returnflag,
        l_linestatus,
        sum(l_quantity) as sum_qty,
        sum(l_extendedprice) as sum_base_price,
        sum(l_extendedprice * (1 - l_discount)) as sum_disc_price,
        sum(l_extendedprice * (1 - l_discount) * (1 + l_tax)) as sum_charge,
        avg(l_quantity) as avg_qty,
        avg(l_extendedprice) as avg_price,
        avg(l_discount) as avg_disc,
        count(*) as count_order
from
        lineitem
where
        l_shipdate <= '1998-9-01'
group by
        l_returnflag,
        l_linestatus
order by
        l_returnflag,
        l_linestatus
                  
```

### CODE REVIEW

**Comment by jinchengchenghh** [comment]
We use this script to build velox, https://github.com/oap-project/gluten/blob/main/tools/build_velox.sh
Maybe you need to build support hdfs

**Comment by KevinyhZou** [comment]
> We use this script to build velox, https://github.com/oap-project/gluten/blob/main/tools/build_velox.sh Maybe you need to build support hdfs

OK, I will try it

**Comment by jinchengchenghh** [comment]
And add some unit tests

**Comment by jinchengchenghh** [comment]
https://github.com/oap-project/gluten/pull/152/files @zhouyuan Can you take eye on this PR? It is similar to yours

File: cpp/velox/compute/DwrfDatasource.cc [hunk 1]
```diff
@@ -121,7 +122,25 @@ std::shared_ptr<arrow::Schema> DwrfDatasource::InspectSchema() {
...
+    std::string hdfsFilePath =
+        regex_replace(file_path_.c_str(), hdfsPrefixExp, "/");
+    HdfsReadFile readFile(hdfs, hdfsFilePath);
+    std::unique_ptr<dwio::common::Reader> reader =
+        dwio::common::getReaderFactory(reader_options.getFileFormat())
+            ->createReader(
+                std::make_unique<dwio::common::FileInputStream>(
+                    file_path_.substr(5)),
+                std::make_unique<dwio::common::ReadFileInputStream>(&readFile),
+                reader_options);
```
Comment (zhouyuan) [inline cpp/velox/compute/DwrfDatasource.cc:141]: createReader() doesn't compile here

```
/home/sparkuser/new_action_runer/actions-runner/_work/gluten/gluten/cpp/velox/compute/DwrfDatasource.cc:141:31: error: no matching function for call to ‘facebook::velox::dwio::common::ReaderFactory::createReader(std::_MakeUniq<facebook::velox::dwio::common::FileInputStream>::__single_object, std::_MakeUniq<facebook::velox::dwio::common::ReadFileInputStream>::__single_object, facebook::velox::dwio::common::ReaderOptions&)’
  141 |                 reader_options);
```

Comment (KevinyhZou) [reply cpp/velox/compute/DwrfDatasource.cc:139]: done.

**Comment by zhouyuan** [comment]
@KevinyhZou Can you please do a rebase? The GHA failures should be fixed in main branch

**Comment by KevinyhZou** [comment]
> @KevinyhZou Can you please do a rebase? The GHA failures should be fixed in main branch

done.

**Review by zhouyuan — APPROVED** [review]

**Review by jinchengchenghh — APPROVED** [review]

**Comment by zhouyuan** [comment]
@KevinyhZou thanks for adding this feature!

**Pull request merged**